tests/integration/telemetry/stackdriver/stackdriver_filter_audit_test.go

```diff
@@ -0,0 +1,161 @@
+// +build integ
+// Copyright Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stackdriver
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/stackdriver"
+	"istio.io/istio/pkg/test/scopes"
+	"istio.io/istio/pkg/test/util/file"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/pkg/test/util/tmpl"
+	"istio.io/istio/tests/integration/telemetry"
+)
+
+const (
+	serverAuditAllLogEntry  = "testdata/security_authz_audit/server_audit_all_log.json.tmpl"
+	serverAuditFooLogEntry  = "testdata/security_authz_audit/server_audit_foo_log.json.tmpl"
+	serverAuditBarLogEntry  = "testdata/security_authz_audit/server_audit_bar_log.json.tmpl"
+	serverAuditNoneLogEntry = "testdata/security_authz_audit/server_audit_none_log.json.tmpl"
+	auditPolicyForLogEntry  = "testdata/security_authz_audit/v1beta1-audit-authorization-policy.yaml.tmpl"
+)
+
+// TestStackdriverAuditLogging testing Authz Policy can config stackdriver with audit policy
+func TestStackdriverHTTPAuditLogging(t *testing.T) {
+	framework.NewTest(t).
+		Features("observability.telemetry.stackdriver").
+		Run(func(ctx framework.TestContext) {
+			g, _ := errgroup.WithContext(context.Background())
+
+			ns := getEchoNamespaceInstance().Name()
+			args := map[string]string{
+				"Namespace": ns,
+			}
+			policies := tmpl.EvaluateAllOrFail(t, args, file.AsStringOrFail(t, auditPolicyForLogEntry))
+			ctx.Config().ApplyYAMLOrFail(t, ns, policies...)
+			defer ctx.Config().DeleteYAMLOrFail(t, ns, policies...)
+			t.Logf("Audit policy deployed to namespace %v", ns)
+
+			t.Logf("Traffic sent to namespace %v", ns)
```

Comment:
Should this log line move to line 70? Or move into `sendTrafficForAudit` fucntion?